pkg/metrics_store/metrics_writer_test.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2021 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metricsstore_test
+
+import (
+	"strings"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/kube-state-metrics/v2/pkg/metric"
+	metricsstore "k8s.io/kube-state-metrics/v2/pkg/metrics_store"
+)
+
+func TestMultiNamespaceMetricsWriter_SingleStores_WriteAll(t *testing.T) {
```

Comment:
Using underscores in function names is against the Golang best practices. Also, since it is a test for the metrics_writer package, it is redundant to mention it in the test name. Usually, you want to name your unit tests `Test<FunctionName>`, so in your case, you could have one main test function called `TestWriteAll` which would execute two sub-tests, one testing single store and another for multiple stores.

Reply:
Good point. I think tests can have different naming conventions depending on the project, but looking at the ones we have, they are all named in PascalCase. 

I renamed the tests in this commit them to match the project's convention, but I wouldn't try to merge them into a single test since the assertions are a bit different and it might just complicate the test. This is also in line with existing tests [1][2].

[1] https://github.com/kubernetes/kube-state-metrics/blob/master/internal/store/endpoint_test.go#L29
[2] https://github.com/kubernetes/kube-state-metrics/blob/master/internal/store/endpoint_test.go#L104